### PR TITLE
feat: expand hardware info coverage

### PR DESCRIPTION
**Pull request by Misieq01**
# #843 #527 

### [ Features ]
- Improved system for implementing reading for various gpu's and cpu's
- Properly display used graphics cards in mining tab
- Added amd graphics cards names to telemetry data
------------
### [ Testing ]
- You should be able to see all of you graphic's cards in mining tab in settings
------------
#### Ubuntu 24.04 | RTX 4080 super | Ryzen 7800x3d
image (2)

#### Windows 11 | RTX 4080 super | Ryzen 7800x3d
image (1)

#### Macbook M1
IMG_3753

#### Windows 11 | RTX 4060 | Intel 12500H
image

### CODE REVIEW

**Review by brianp — APPROVED** [review]

File: src-tauri/src/hardware/gpu_readers/amd_gpu_reader.rs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+use anyhow::Error;
+use async_trait::async_trait;
+// use libamdgpu_top::{
+//     DevicePath,
+//     AMDGPU::{HwmonTemp, PowerCapType},
+// };
+// use libdrm_amdgpu_sys::AMDGPU::{
+//     get_all_amdgpu_pci_bus, DeviceHandle, HwmonTempType, PowerCap, GPU_INFO, HW_IP::HW_IP_TYPE,
+//     SENSOR_INFO::SENSOR_TYPE,
+// };
```
[the remark this answers is not in the document]
Comment (Misieq01) [reply src-tauri/src/hardware/gpu_readers/amd_gpu_reader.rs:12]: Good catch

File: src-tauri/src/hardware/gpu_readers/amd_gpu_reader.rs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        // info!("GPU count: {}", gpu_list.size());
+
+        // for gpu in 0..gpu_list.size() {
+        //     let gpu = gpu_list.at(gpu)?;
+        //     let gpu1 = gpu.cast::<Gpu1>()?;
+        //     info!("GPU name: {}", gpu1.name()?);
+        //     info!("GPU product name: {}", gpu1.product_name()?)
+        // };
+        // let test = get_all_amdgpu_pci_bus();
+
```
[the remark this answers is not in the document]
Comment (Misieq01) [reply src-tauri/src/hardware/gpu_readers/amd_gpu_reader.rs:66]: I was thinking about leaving it as kind of documentation about what have been tried but I can delete it

[the remark this answers is not in the document]
Comment (Misieq01) [reply src-tauri/src/hardware/gpu_readers/amd_gpu_reader.rs:66]: Had to remove parameters reading for amd for now as it had problem with crossplatform support e.g

**Review by brianp — APPROVED** [review]

**Comment by Anziskavds** [comment]
Tested successfully on MacBook Air 13 inch M1 2020 and on a Acer Aspire A315-59 on Windows 11 on Testnet v0.6.4.

**Comment by Tas4tari** [comment]
Tested successfully on Windows 11, app version 0.6.4 whereby my GPU devices are now listed:
Screenshot (251)

However issue persists where Mining widget shows GPU Power as null, adding comments to:
https://github.com/tari-project/universe/issues/843

**Pull request merged**